tensorboard/components/tensor_widget/selection.ts

```diff
@@ -0,0 +1,339 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {numElements} from './shape-utils';
+import {Shape, TensorViewSlicingSpec} from './types';
+
+/**
+ * The possible status of a selected cell.
+ */
+export enum CellSelectionStatus {
```

Comment:
A given cell can have any of these properties.  Instead of `CellSelectionStatus[]`, I think it would be more natural to represent that as as interface with five booleans.  If you want to get really precise about the edge states being valid only if the cell is selected, you could have something like `type CellSelectionStatus = null | {leftEdge?: boolean, rightEdge?: boolean ...}`.  Then `null` would mean 'not selected' and `{}` would mean 'selected but not on an edge'.

Reply:
Great point. I've adopted your "precise" option.